src/client/mesh_generator_thread.cpp

```diff
@@ -124,6 +124,18 @@ bool MeshUpdateQueue::addBlock(Map *map, v3s16 p, bool ack_block_to_server,
 		}
 	}
 
+	/*
+		Air blocks won't suddenly become visible due to a neighbor update, so
+		skip those.
+		Note: this can be extended with more precise checks in the future
+	*/
+	if (from_neighbor && mesh_grid.cell_size == 1 && main_block->isAir()) {
+		assert(!ack_block_to_server);
+		m_urgents.erase(mesh_position);
+		g_profiler->add("MeshUpdateQueue: updates skipped", 1);
+		return true;
+	}
```

Comment:
Why not do this check further up, before updating the queued entry?
(If it's air, and it's in the queue, it must have been non-air before. But then, there would be a non-neighbor call for it, because its nodes changed. So I don't see a reason for updating the queue entry.)